iocore/net/SSLUtils.cc

```diff
@@ -344,6 +361,11 @@ set_context_cert(SSL *ssl)
       ctx = cc->ctx;
   }
 
+  if (cc && cc->hpkp) {
+    netvc->hpkp = cc->hpkp;
```

Comment:
in that case, you are racing against configuration reload and will dereference a bad pointer if the transaction takes a long time.